### ISSUE TITLE
webdriverio v > 8.13 breaks wdio-electron-service

### ISSUE THREAD

**Issue by ForesightImaging** [webdriverio-community/wdio-electron-service#167]
I can run with webdriverio v8.13, but in later versions the chrome driver isn't resolved correctly.

I noticed that as of 8.14 [wdio-chromdriver-service is deprecated and no longer maintained](https://github.com/webdriverio-community/wdio-chromedriver-service). There's also a [blog post](https://webdriver.io/blog/2023/07/31/driver-management) about how 8.14 handles chrome driver resolution for you.

Passing an electronVersion is ignored as is passing a chromedriverCustomPath. The runner located the electron executable and refuses to run because the driver it has selected (for the installed chrome) isn't a match.

I can run with 8.13, but after, including 8.15 seems DOA due to upstream changes.

**Comment by christian-bromann** [comment]
Thanks for reporting!



We greatly appreciate any contributions that help resolve the bug. While we understand that active contributors have their own priorities, we kindly request your assistance if you rely on this bug being fixed. We encourage you to take a look at our [contribution guidelines](https://webdriver.io/docs/contribute) or join our friendly [Discord](http://discord.webdriver.io/) development server, where you can ask any questions you may have. Thank you for your support, and cheers!

**Comment by goosewobbler** [comment]
Progress for this can be seen in #168.  In the meantime, the workaround is simply to use WDIO v8.13

**Comment by ForesightImaging** [comment]
we can't use 8.13 as it has problems with custom reporters.

**Comment by christian-bromann** [comment]
> it has problems with custom reporters.

What problems?

**Comment by ForesightImaging** [comment]
the [problems with mixed module types that you recently fixed](https://github.com/webdriverio/webdriverio/pull/10998). That's not in 8.13. So I can't fall back to 8.13.

**Comment by ForesightImaging** [comment]
FWIW, my recent journey started with a webdriver update and has led me through a lot of ESM and CJS module trouble. Some environments, like [electron are not ready for ESM](https://github.com/electron/electron/pull/37535) just yet. Every nook and cranny has its own trouble. For example as I got out jest test, webdriver tests, webdriver config, electron main process, renderer process and webpack config aligned on a single .tsconfig and module format I have found that every tool has quirks and expectations compatibility tools like [esm even lack support for optional chaining](https://github.com/standard-things/esm/issues/866). So supporting CJS is going to help others with the transition.
